phrase_extraction_evaluation.py

```diff
@@ -0,0 +1,92 @@
+# Class to evaluate model output
+
+import numpy as np
+from typing import List
+from sentence_transformers import SentenceTransformer
+
+embedding_model = SentenceTransformer("all-MiniLM-L6-v2")
+
+class STSEvaluation:
+    treshhold=0.75
+
+    @staticmethod
+    def get_embedding(phrase: str) -> np.ndarray:
+        return embedding_model.encode([phrase])[0]
+
+    @staticmethod
+    def get_embeddings(phrases: List[str]) -> List[np.ndarray]:
+        return embedding_model.encode(phrases)
+    
+    @staticmethod
+    def cosine_similarity(a: np.ndarray, b: np.ndarray) -> float:
+        dot_product = np.dot(a, b)
+        a_magnitude = np.linalg.norm(a)
+        b_magnitude = np.linalg.norm(b)
+        if a_magnitude == 0 or b_magnitude == 0:
+            return 0.0
+        return dot_product / (a_magnitude * b_magnitude)
+    
+    @staticmethod
+    def make_similarity_matrix(predictions, labels):
```

Comment:
Evo jedne fore kako se u python-u pisu ovakve stvari krace, a zbog toga kako python funkcionise budu i brze. Razmisli da li mozes jos negde ovo primeniti:

def make_similarity_matrix(predictions, labels):
    return np.array([[STSEvaluation.cosine_similarity(p, l) for l in labels] for p in predictions])